valhalla/jawn/src/managers/traceManager.ts

```diff
@@ -0,0 +1,225 @@
+import type { Log, Message } from "../lib/handlers/HandlerContext";
+import { KafkaProducer } from "../lib/clients/KafkaProducer";
+import { AuthParams } from "../lib/db/supabase";
+import { S3Client } from "../lib/shared/db/s3Client";
+
+import { randomUUID } from "crypto";
+import { parseInt } from "lodash";
+
+export class TraceManager {
+
+  private s3Client: S3Client;
+  constructor() {
+    this.s3Client = new S3Client(
+      process.env.S3_ACCESS_KEY ?? "",
+      process.env.S3_SECRET_KEY ?? "",
+      process.env.S3_ENDPOINT ?? "",
+      process.env.S3_BUCKET_NAME ?? "",
+      (process.env.S3_REGION as "us-west-2" | "eu-west-1") ?? "us-west-2"
+    );
+  }
+
+  public async consumeTraces(trace: OTELTrace, heliconeAuthorization: string, authParams: AuthParams) {
```

Comment:
Hi @YashKarthik 

 Can we make this function less imperative and more functional? Especially for this type of parsing, we want to keep it easier to read and debug in the future